bench/benchmarks/layout.js

```diff
@@ -66,41 +66,39 @@ export default class Layout extends Benchmark {
                 this.glyphs = {};
                 this.icons = {};
 
-                const preloadGlyphs = (params, callback) => {
-                    style.getGlyphs('', params, (err, glyphs) => {
-                        this.glyphs[JSON.stringify(params)] = glyphs;
-                        callback(err, glyphs);
-                    });
-                };
-
                 const preloadImages = (params, callback) => {
                     style.getImages('', params, (err, icons) => {
                         this.icons[JSON.stringify(params)] = icons;
                         callback(err, icons);
                     });
                 };
 
-                return this.bench(preloadGlyphs, preloadImages);
+                const preloadGlyphs = (params, callback) => {
+                    style.getGlyphs('', params, (err, glyphs) => {
+                        this.glyphs[JSON.stringify(params)] = glyphs;
+                        callback(err, glyphs);
+                    });
+                };
+
+                return this.parseTiles(preloadImages, preloadGlyphs);
             });
     }
 
-    bench(getGlyphs: Function = (params, callback) => callback(null, this.glyphs[JSON.stringify(params)]),
-          getImages: Function = (params, callback) => callback(null, this.icons[JSON.stringify(params)])) {
+    parseTiles(loadImages: Function, loadGlyphs: Function) {
```

Comment:
Rather than using inheritance for code reuse, could we pull `parseTiles` out into a module in `bench/lib` that is used by `Layout` and `WorkerTransfer`?

Reply:
We could, but I think I'd prefer inheritance in this case — it's easier when all the actual benchmark code is in one folder, and we might want to customize `parseTiles` in other benchmarks. `LayoutDDS` also inherits `Layout` and rewriting it to use composition or util methods for consistency will be very cumbersome.